src/synthesizer/blackholes.py

```diff
@@ -0,0 +1,460 @@
+                 hydrogen_density_blr=1E9/cm**3,
+                 covering_fraction_blr=0.1,
+                 velocity_dispersion_blr=2000*km/s,
+                 ionisation_parameter_nlr=0.01,
+                 hydrogen_density_nlr=1E4/cm**3,
+                 covering_fraction_nlr=0.1,
+                 velocity_dispersion_nlr=500*km/s,
+                 theta_torus=10*deg,
+                 torus_emission_model=Greybody(1000*K, 1.5),
+                 ):
+
+        """
+        Arguments:
+
+            disc_model (str)
+                The disc_model to be used. The current test model is the AGNSED
+                model.
+            photoionisation_model (str)
+                The photoionisation model to be used. Normally this would be 
+                e.g. "cloudy_c17.03", for the test_grid this is "" (i.e. 
+                empty string).
+            grid_dir (str)
+                The path to the grid directory.
+            bolometric_luminosity (float)
+                This is needed to rescale the spectra since the grid is likely
+                to be coarse. Defaults to None since it should be provided.
+            metallicity (float)
+                The metallicity of the NLR and BLR gas. Defaults to None since
+                it should be provided.
+            ionisation_parameter_blr (float)
+                The  ionisation parameter in the BLR. Default value
+                is 0.1.
+            hydrogen_density_blr (unyt.unit_object.Unit)
+                The  hydrogen density in the BLR. Default value
+                is 1E9/cm**3.
+            covering_fraction_blr (float)
+                The covering fraction of the BLR. Default value is 0.1.
+            velocity_dispersion_blr (unyt.unit_object.Unit)
+                The velocity disperson of the BLR. Default value is 
+                2000*km/s.
+            ionisation_parameter_nlr (float)
+                The ionisation parameter in the BLR. Default value
+                is 0.01.
+            hydrogen_density_nlr (unyt.unit_object.Unit)
+                The hydrogen density in the NLR. Default value
+                is 1E4/cm**3.
+            covering_fraction_nlr (float)
+                The covering fraction of the NLR. Default value is 0.1.
+            velocity_dispersion_nlr (unyt.unit_object.Unit)
+                The velocity disperson of the NLR. Default value is 
+                500*km/s.
+            theta_torus (float)
+                The opening angle of the torus component. Default value is
+                10*deg.
+            torus_emission_model (synthesizer.dust.emission)
+                The torus emission model. A synthesizer dust emission model.
+                Default is a Greybody with T=1000*K and emissivity=1.6.
+            **kwargs 
+                Addition parameters for the specific disc model.
+
+        """
+
+        # Create a dictionary of all the arguments (and self)
+        args = locals()
```

Comment:
Not sure I'm a fan of this. `locals` is generally frowned upon. What's wrong with just hard coding the dictionary from the arguments? Then its fully documented. More typing sure, but much cleaner.

Reply:
Done